hawc/apps/assessment/views.py

```diff
@@ -828,6 +848,41 @@ def get_context_data(self, **kwargs):
         return context
 
 
+# communications
+
+
+class CommunicationUpdate(UpdateView):
+    template_name = "assessment/communication_update.html"
+    model = models.Communication
+    form_class = forms.CommunicationForm
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        content_type = ContentType.objects.get(app_label="assessment", model="assessment")
+        assessment = content_type.get_object_for_this_type(id=(self.object.object_id))
+        context["assessment"] = assessment
+        context["breadcrumbs"] = Breadcrumb.build_crumbs(
+            self.request.user, "Communication Update", extras=[Breadcrumb.from_object(assessment)],
+        )
+        return context
+
+    def get_object(self, queryset=None):
+        content_type = ContentType.objects.get(app_label="assessment", model="communication")
+        obj, created = models.Communication.objects.get_or_create(
+            content_type=content_type, object_id=self.kwargs["pk"]
+        )
+        return obj
+
+    def post(self, request, *args, **kwargs):
```

Comment:
try removing this and see if the complete lifecycle works; I don't think we need it.  The only thing you may need to change is to add a `get_success_url` method instead, and add the `reverse("assessment:detail", args=[kwargs["pk"]]`